src/mmap_unix.rs

```diff
@@ -74,6 +74,156 @@ impl error::Error for Error {}
 
 pub type Result<T> = result::Result<T, Error>;
 
+/// A factory struct to build `MmapRegion` objects.
+pub struct MmapRegionBuilder<B = ()> {
+    size: usize,
+    prot: i32,
+    flags: i32,
+    file_offset: Option<FileOffset>,
+    raw_ptr: Option<*mut u8>,
+    hugetlbfs: Option<bool>,
+    bitmap: B,
+}
+
+impl<B: Bitmap + NewBitmap> MmapRegionBuilder<B> {
+    /// Create a new MmapRegionBuilder.
+    pub fn new(size: usize) -> Self {
+        MmapRegionBuilder {
+            size,
+            prot: libc::PROT_READ | libc::PROT_WRITE,
+            flags: libc::MAP_ANONYMOUS | libc::MAP_NORESERVE | libc::MAP_PRIVATE,
```

Comment:
Is it expected for these fields to have this value initially?

Reply:
good suggestion, setting `prot` and `flags` to zero simplifies the implementations:)